node/api.go

```diff
@@ -58,14 +59,30 @@ func (api *PrivateAdminAPI) AddPeer(url string) (bool, error) {
 }
 
 // StartRPC starts the HTTP RPC API server.
-func (api *PrivateAdminAPI) StartRPC(host string, port int, cors string, apis string) (bool, error) {
+func (api *PrivateAdminAPI) StartRPC(host *string, port *rpc.HexNumber, cors *string, apis *string) (bool, error) {
```

Comment:
The reason is that with pointers the RPC package is able to detect that a parameters is optional. If there is no value for this parameter given by the client the RPC package can supply nil to the callback as an indication the client has not specified the value (or used json null). The callback can then act accordingly. In this case use default values.

If the callback uses a value instead of a pointer the RPC package assumes that a value must be given by the client. If not it returns an error that parameters are missing. It cannot use the default value for a type because the callback is then not able to distinct between null/not specified and the default value. In this case that might be fine since the empty value is the same value as the default value, but in general this might not be the case. For example in the case of an integer which can be omitted by the client. If the RPC layer uses the default 0 value this can be an actually meaningful value while the client didn't specified the value.